packages/vite/src/node/server/middlewares/static.ts

```diff
@@ -112,28 +106,12 @@ export function serveRawFsMiddleware(
   }
 }
 
-function renderFsRestrictedHTML(root: string) {
-  // to have syntax highlighting and autocompletion in IDE
-  const html = String.raw
-  return html`
-    <body>
-      <h1>403 Restricted</h1>
-      <p>
-        For security concerns, accessing files outside of workspace root
-        (<code>${root}</code>) is restricted since Vite v2.3.x
-      </p>
-      <p>
-        Refer to docs
-        <a href="https://vitejs.dev/config/#server-fsserveroot">
-          https://vitejs.dev/config/#server-fsserveroot
-        </a>
-        for configurations and more details.
-      </p>
-      <style>
-        body {
-          padding: 1em 2em;
-        }
-      </style>
-    </body>
-  `
+export function checkFileOutSide(url: string, serveRoot: string): void {
+  if (!url.startsWith(serveRoot + path.sep)) {
+    throw new FileOutSideError(
+      `The request url "${url}" is outside of vite dev server root "${serveRoot}". \nFor security concerns, accessing files outside of workspace root is restricted since Vite v2.3.x. \nRefer to docs https://vitejs.dev/config/#server-fsserveroot for configurations and more details.`,
```

Comment:
Maybe break this down to multi-line for maintainability?

Reply:
I replaced `\n` to `<br/>` at html.
``` js
export function renderErrorHTML(msg: string): string {
  // to have syntax highlighting and autocompletion in IDE
  const html = String.raw
  return html`
    <body>
      <h1>403 Restricted</h1>
      <p>${msg.replace(/\n/g, '<br/>')}</p>
      <style>
        body {
          padding: 1em 2em;
        }
      </style>
    </body>
  `
}
```

Reply:
I mean, like this 

```ts
`The request url "${url}" is outside of vite dev server root "${serveRoot}". 
For security concerns, accessing files outside of workspace root is restricted since Vite v2.3.x. 

Refer to docs https://vitejs.dev/config/#server-fsserveroot for configurations and more details.`,
```